src/nbl/video/utilities/CPropertyPoolHandler.cpp

```diff
@@ -5,11 +5,36 @@
 using namespace nbl;
 using namespace video;
 
-#if 0 // TODO: port
 //
-CPropertyPoolHandler::CPropertyPoolHandler(core::smart_refctd_ptr<ILogicalDevice>&& device) : m_device(std::move(device)), m_dsCache()
+CPropertyPoolHandler::CPropertyPoolHandler(core::smart_refctd_ptr<ILogicalDevice>&& device) : m_device(std::move(device))
 {
-	// TODO: rewrite in HLSL!
+	auto system = m_device->getPhysicalDevice()->getSystem();
+	// TODO: Reuse asset manager from elsewhere?
+	auto assetManager = core::make_smart_refctd_ptr<asset::IAssetManager>(core::smart_refctd_ptr<system::ISystem>(system));
+
+	auto loadShader = [&](const char* path)
+		{
+			asset::IAssetLoader::SAssetLoadParams params = {};
+			auto assetBundle = assetManager->getAsset(path, params);
+			auto assets = assetBundle.getContents();
+			assert(!assets.empty());
+
+			auto cpuShader = asset::IAsset::castDown<asset::ICPUShader>(assets[0]);
+			auto shader = m_device->createShader(cpuShader.get());
+			return shader;
+		};
+	auto shader = loadShader("../../../include/nbl/builtin/hlsl/property_pool/copy.comp.hlsl");
```

Comment:
I'd rather use `system->openFile` and use the `nbl/builtin/hlsl/property_pool/copy.comp.hlsl` path directly toi opena  memory mapped file and create `IGPUShader` from it directly

Reply:
look at the old `loadBuiltinData` code